src/uu/split/src/split.rs

```diff
@@ -499,12 +517,29 @@ impl Settings {
                 match first.as_str() {
                     "\\0" => b'\0',
                     s if s.as_bytes().len() == 1 => s.as_bytes()[0],
-                    s => return Err(SettingsError::MultiCharacterSeparator(s.to_owned())),
+                    s => return Err(SettingsError::MultiCharacterSeparator(s.to_string())),
                 }
             }
             None => b'\n',
         };
 
+        let io_blksize: Option<usize> = if let Some(s) = matches.get_one::<String>(OPT_IO_BLKSIZE) {
```

Comment:
What do you think about ?
let io_blksize: Option<usize> = matches.get_one::<String>(OPT_IO_BLKSIZE).map(|s| {
    parse_size_u64(s)
        .ok()
        .and_then(|n| {
            let n = n.try_into().map_err(|_| SettingsError::InvalidIOBlockSize(s.to_string()))?;
            if n > OPT_IO_BLKSIZE_MAX {
                Err(SettingsError::InvalidIOBlockSize(s.to_string()))
            } else {
                Ok(n)
            }
        })
}).transpose()?;

Reply:
in this case it would "eat" possible error returned by parse_size_u64(), so it cannot be bubbled up to from()->Result. Also, since ok() converts it to Option and the and_then() expects an option to be returned from the closure, it is not straitforward to return the SettingsError in either places within the closure inside closure - i.e. doing `return Err(SettingsError::InvalidIOBlockSize(s.to_string()))` or `Some(Err(SettingsError::InvalidIOBlockSize(s.to_string())))` would not work : as return is not for from()->Result function but closure, and wrapping in Some() to satisfy and_then() signature we will end up with `Option<Option<Result<>>>` as a retuning value from closure from map() instead of `Option<Result<>>` ... could unwrap(), but then still missing/eating error from parse_size_u64() ... 
I could be missing something though

Reply:
please ignore my comment then :(